src/etc/inc/captiveportal.inc

```diff
@@ -2525,6 +2580,9 @@ function portal_allow($clientip, $clientmac, $username, $password = null, $attri
 		write_config(gettext("Captive Portal allowed users configuration changed"));
 	}
 
+	if ($existing_sessionid !== 'null' && !empty($sessionid)) {
```

Comment:
The first test here needs its own set of parenthesis

Reply:
Do you mean something like ```if (($existing_sessionid !== 'null') && !empty($sessionid)) {``` ?

Reply:
Yes